### PR TITLE
fix(processing): convert tensor to numpy array before np.cumsum in processors (#48177)

### LINKED ISSUE #48177: InternVL processor passes a torch tensor to `np.cumsum`, emitting a NumPy `__array_wrap__` DeprecationWarning
### System Info

- `transformers` 5.15.1 (also reproduced on `main`)
- `torch` 2.13.0
- `numpy` 2.5.2
- Python 3.12

### Who can help?

Anyone maintaining the InternVL / GOT-OCR2 processors.

### Description

`InternVLProcessor.__call__` passes a `torch.Tensor` to `np.cumsum`, which makes NumPy fall back to its legacy `__array_wrap__` path and emit a `DeprecationWarning` on every call that receives images.

`src/transformers/models/internvl/processing_internvl.py`, [L110](https://github.com/huggingface/transformers/blob/v5.15.1/src/transformers/models/internvl/processing_internvl.py#L110):

```python
if images is not None:
    image_num_patches_indices = np.cumsum(model_inputs.pop("num_patches"))
```

`num_patches` is a `list[int]` inside the image processor, but it is returned through a `BatchFeature` built with `tensor_type=return_tensors` — `src/transformers/models/got_ocr2/image_processing_got_ocr2.py`, [L267](https://github.com/huggingface/transformers/blob/v5.15.1/src/transformers/models/got_ocr2/image_processing_got_ocr2.py#L267):

```python
return BatchFeature(
    data={"pixel_values": processed_images, "num_patches": num_patches}, tensor_type=return_tensors
)
```

so with the usual `return_tensors="pt"` it reaches `np.cumsum` as a tensor.

`np.cumsum` on a non-ndarray goes through NumPy's `_wrapfunc`, which tries `obj.cumsum(axis=…, dtype=…, out=…)`; `Tensor.cumsum` rejects those keywords with a `TypeError`, so NumPy falls back to `_wrapit` and wraps the result via `type(obj).__array_wrap__`. `torch.Tensor.__array_wrap__(self, array)` still has the pre-NumPy-2.0 two-argument signature ([pytorch#180657](https://github.com/pytorch/pytorch/issues/180657), open PR [pytorch#187250](https://github.com/pytorch/pytorch/pull/187250)), so NumPy warns.

The warning is a `DeprecationWarning`, so it is normally invisible; it surfaces in test suites, which is how we found it — it is emitted once per processor call in `huggingface/trl`'s InternVL tests (126 emissions in one CI run, https://github.com/huggingface/trl/issues/6848).

Beyond the warning, the round-trip through NumPy is unnecessary here: the tensor is converted to an `ndarray`, and `image_num_patches_indices` is then only indexed to slice `model_inputs["pixel_values"]`.

### Reproduction

Call `InternVLProcessor` with images and `return_tensors="pt"`, with `DeprecationWarning`s made visible (`python -W error::DeprecationWarning`, or under `pytest`, which shows them by default).

Observed in `huggingface/trl`'s CI, where the processor is called on `trl-internal-testing/tiny-InternVLForConditionalGeneration`:

```
tests/test_distillation_trainer.py: 24 warnings
tests/test_grpo_trainer.py: 51 warnings
tests/test_rloo_trainer.py: 51 warnings
  .venv/lib/python3.12/site-packages/numpy/_core/fromnumeric.py:45: DeprecationWarning: __array_wrap__ must accept context and return_scalar arguments (positionally) in the future. (Deprecated NumPy 2.0)
    return conv.wrap(result, to_scalar=False)
```

Full log: https://github.com/huggingface/trl/actions/runs/32414561958/job/96572632322 — the emissions come from the `test_train_vlm[trl-internal-testing/tiny-InternVLForConditionalGeneration]` cases of those three files, one per processor call.

The last step in isolation, without any model:

```python
>>> import numpy as np, torch, warnings
>>> warnings.simplefilter("error")
>>> np.cumsum(torch.tensor([1, 2, 3]))
DeprecationWarning: __array_wrap__ must accept context and return_scalar arguments (positionally) in the future. (Deprecated NumPy 2.0)
```

### Expected behavior

No warning. Computing the offsets without the NumPy round-trip fixes it, e.g. by keeping the cumulative sum in torch, or by taking it over the values before they are tensorized. Both keep the subsequent slicing unchanged, since only `image_num_patches_indices[i]` is used.

The same pattern appears in two other processors, where the operand also comes out of a `BatchFeature` and would therefore be a tensor with `return_tensors="pt"` — I have not verified these, flagging them in case they are worth fixing together:

- `models/minicpmv4_6/processing_minicpmv4_6.py:151` — `np.cumsum(image_inputs["num_patches_per_image"])`
- `models/kimi_k25/processing_kimi_k25.py:79` — `np.cumsum(video_inputs["num_chunks_per_video"])`

(`models/got_ocr2/processing_got_ocr2.py:237` is fine — it sums a plain Python list.)

Happy to send a PR for InternVL, and for the other two if you want them in the same change.

### PR DESCRIPTION
**Pull request by loulanyue**
[CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=48179)

## Summary

Fixes #48177.

When multimodal processors like `InternVLProcessor`, `MiniCPMV4_6Processor`, and `Kimi_K25Processor` process batches with `return_tensors="pt"`, slice/patch counts (such as `num_patches`, `num_patches_per_image`, `num_frames_per_video`, `num_chunks_per_video`) are returned as PyTorch tensors.

Passing a `torch.Tensor` directly into `np.cumsum` causes NumPy to fall back to `_wrapit` and invoke `torch.Tensor.__array_wrap__`, triggering a `DeprecationWarning: __array_wrap__ must accept context and return_scalar arguments (positionally) in the future. (Deprecated NumPy 2.0)` on every call that processes inputs with images/videos.

## Changes

- Use `to_numpy(...)` on the tensor operands before calling `np.cumsum(...)` in:
  - `src/transformers/models/internvl/processing_internvl.py`
  - `src/transformers/models/minicpmv4_6/processing_minicpmv4_6.py`
  - `src/transformers/models/kimi_k25/modular_kimi_k25.py` and `src/transformers/models/kimi_k25/processing_kimi_k25.py`

### CODE REVIEW

**Comment by Rocketknight1** [comment]
Please don't randomly run your code agent on people's issues! It just creates noise, and people are able to run code agents themselves, they don't need you to do it for them
